dango/testing/tests/hyperlane_va.rs

```diff
@@ -0,0 +1,304 @@
+        let message_hash = eip191_hash(announcement_hash(
+            domain_hash(local_domain, mailbox.into(), VA_DOMAIN_KEY),
+            storage_location,
+        ));
+
+        let (signature, recovery_id) = sk
+            .sign_digest_recoverable(Identity256::from(message_hash.into_inner()))
+            .unwrap();
+
+        let mut packed = [0u8; 65];
+        packed[..64].copy_from_slice(&signature.to_bytes());
+        packed[64] = recovery_id.to_byte() + 27;
+
+        Self {
+            sk,
+            validator: validator_address.try_into().unwrap(),
+            signature: packed.into(),
+            storage_location: storage_location.to_string(),
+        }
+    }
+}
+
+#[test]
+fn test_announce() {
+    let (mut suite, mut accounts, _, contracts) = setup_test();
+
+    let mut signer = accounts.user1;
+
+    let mailbox = contracts.hyperlane.mailbox;
+    let local_domain = suite
+        .query_wasm_smart(mailbox, mailbox::QueryConfigRequest {})
+        .should_succeed()
+        .local_domain;
+
+    let va = contracts.hyperlane.va;
+
+    let mut validators_expected = BTreeSet::new();
+    let mut storage_locations_expected = BTreeMap::new();
+
+    // Create a working announcement.
+    let announcement = MockAnnouncement::new(mailbox, local_domain, "Test/Storage/Location");
+
+    // Adding a valid announcement.
+    {
+        suite
+            .execute(
+                &mut signer,
+                va,
+                &va::ExecuteMsg::Announce {
+                    storage_location: announcement.storage_location.clone(),
+                    validator: announcement.validator,
+                    signature: announcement.signature,
+                },
+                Coins::new(),
+            )
+            .should_succeed()
+            .events
+            .search_event::<ContractEvent>()
+            .with_predicate(|evt| evt.ty == "validator_announcement")
+            .take()
+            .one()
+            .event
+            .data
+            .deserialize_json::<Announce>()
+            .should_succeed_and_equal(Announce {
+                sender: signer.address(),
+                validator: announcement.validator,
+                storage_location: announcement.storage_location.clone(),
+            });
+
+        // Check that the validator was added to the validators.
+        validators_expected.insert(announcement.validator);
+
+        suite
+            .query_wasm_smart(va, va::QueryAnnouncedValidatorsRequest {})
+            .should_succeed_and_equal(validators_expected.clone());
+
+        // Check that the validator was added to the storage locations.
+        storage_locations_expected.insert(
+            announcement.validator,
+            BTreeSet::from(["Test/Storage/Location".to_string()]),
+        );
+
+        suite
+            .query_wasm_smart(va, va::QueryAnnounceStorageLocationsRequest {
+                validators: BTreeSet::from_iter([announcement.validator]),
```

Comment:
To create BTreeSet, use our dedicated macro
```suggestion
                validators: btree_set![announcement.validator]
```